ir/src/constraint_builder/boundary_constraints.rs

```diff
@@ -71,8 +71,9 @@ impl ConstraintBuilder {
                    "The boundary constraint's domain should be {lhs_domain:?}, but the domain {domain:?} was inferred by the graph",
                );
 
+                let (_, _, value) = constraint.into_parts();
```

Comment:
We should do the `into_parts` call at the top of the case and then use the access and boundary, instead of calling `constraint.access()` and `constraint.boundary()` like above

```
                let (boundary, access, value) = constraint.into_parts();
                let trace_access = self.symbol_table.get_trace_binding_access(&access)?;
                let domain = boundary.into();
```